ui/__init__.py

```diff
@@ -51,45 +57,73 @@ def _select_cell(self, cell):
         if move:
             self._apply_move(move)
 
-    def _apply_move(self, move):
+    def _process_agent_move(self):
         """
-        Applies the given move to the game board, updating both the model and
-        view accordingly.
-        :param move: the Move to apply
+        Hands control over to the agent to perform a move if the player to move
+        is CPU-controlled.
         :return: None
         """
-        self._view.apply_move(move, board=self._model.game_board, on_end=self._process_agent_move)
-        self._model.apply_move(move)
+        config = self._model.game_config
+        player_color = self._model.game_turn
+        player_type = config.get_player_type(player_color)
 
-    def _process_agent_move(self):
+        if player_type is PlayerType.COMPUTER:
+            self._agent_operator.search(self._model.game_board,
+                                        player_color,
+                                        self._set_timeout_move)
```

Comment:
The agent now searches calling _set_timeout_move every time a new move is found.
Currently this is called repeatedly with random moves, but when heuristic is completed, this would be called every time a move with a better heuristic value is found (Best current move)